### PR TITLE
Add cachefor to Azure Storage persistence 

### PR DESCRIPTION
**Pull request by SimonCropp**
WIP until 1.3 is released

### CODE REVIEW

**Comment by SimonCropp** [comment]
ping @Particular/azure-maintainers

File: nservicebus/azure-storage-persistence/configuration.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ The following settings are available for changing the behavior of subscription p
   * defaults to `true`.
  * `TableName`: Specifies the name of the table for storing subscriptions
   * defaults to `Subscription`.
+ * `CacheFor`: (Added in Version 1.3) By default every time a message is published the subscription storage is queried. In scenarios where the list of subscribers rarely changes this query is often redundant and can slow down message processing. `CacheFor` allows subscriptions to be cached for a given period of time, hence helping reduce the impact of redundant queries to the subscription store.
```
Comment (SeanFeldman) [inline nservicebus/azure-storage-persistence/configuration.md]: What's the default value?

Comment (SimonCropp) [inline nservicebus/azure-storage-persistence/configuration.md]: " By default every time a message is published the subscription storage is queried"

Comment (SeanFeldman) [inline nservicebus/azure-storage-persistence/configuration.md]: Attention to details on my end 😆

Comment (SimonCropp) [inline nservicebus/azure-storage-persistence/configuration.md]: happy to word it better i u can think of a way. 

BTW if this was added in a major i would make it a required setting.

**Comment by danielmarbach** [comment]
Smoked tested Sample with ASP 1.3

**Pull request merged**